cloud/ec2.go

```diff
@@ -218,12 +218,6 @@ func (m *ec2Manager) spawnOnDemandHost(ctx context.Context, h *host.Host, ec2Set
 			"host_provider": h.Distro.Provider,
 			"distro":        h.Distro.Id,
 		}))
-		grip.Error(message.WrapError(h.Remove(), message.Fields{
```

Comment:
This call to `(*host.Host).Remove()` does not seem correct, since it should be removed in `createHostJob`. Same for the call in `spawnSpotHost()`.

Reply:
I'm not sure it matters much where it gets called. The concern here, I think, is that if we don't clean up here, do we potentially leave something in the database for longer than we should and would that cause problems?

Reply:
I think the question is, will the job actually remove the intent host if it returns here? https://github.com/evergreen-ci/evergreen/blob/37dbd10f5aa1784aac6d5511e781e7e1b209ea6b/units/provisioning_create_host.go#L216